Description: Disable this optimisation until LLVM-7 has libLLVMPolly etc
Author: Alastair McKinstry <mckinstry@debian.org>
Last-Updated: 2018-12-03
Forwarded: no

Index: flang-20181226/flang-driver/tools/driver/cc1_main.cpp
===================================================================
--- flang-20181226.orig/flang-driver/tools/driver/cc1_main.cpp
+++ flang-20181226/flang-driver/tools/driver/cc1_main.cpp
@@ -67,11 +67,6 @@ static void LLVMErrorHandler(void *UserD
   exit(GenCrashDiag ? 70 : 1);
 }
 
-#ifdef LINK_POLLY_INTO_TOOLS
-namespace polly {
-void initializePollyPasses(llvm::PassRegistry &Registry);
-}
-#endif
 
 #ifdef CLANG_HAVE_RLIMITS
 #if defined(__linux__) && defined(__PIE__)
@@ -181,10 +176,7 @@ int cc1_main(ArrayRef<const char *> Argv
   llvm::InitializeAllAsmPrinters();
   llvm::InitializeAllAsmParsers();
 
-#ifdef LINK_POLLY_INTO_TOOLS
-  llvm::PassRegistry &Registry = *llvm::PassRegistry::getPassRegistry();
-  polly::initializePollyPasses(Registry);
-#endif
+
 
   // Buffer diagnostics from argument parsing so that we can output them using a
   // well formed diagnostic object.
